Moving to level 3

Kjusok opened the pull request.

Kjusok committed:
WIP TestToonShader

Kjusok committed:
Try Moving to level 3

Try Moving to level 3

Kjusok committed:
Fixed enemy spawn point

Fixed enemy spawn point

Kjusok committed:
Merge branch 'feature/AddedAssets_1' into feature/TryMovingToLevel3

Kjusok committed:
Added player model and animations

Added player model and animations

Kjusok committed:
Added starting weapons + effects for the player

Added starting weapons + effects for the player

Kjusok committed:
Added 4 types of weapons + shot and hit effects

Added 4 types of weapons + shot and hit effects

Kjusok committed:
Merge branch 'develop' into feature/TryMovingToLevel3

ega-ega commented on Assets/Scripts/Gameplay/Enemy/Spawners/EnemySpawner.cs:
```diff
@@ -1,5 +1,7 @@
+using Gameplay.AI;
```
pls remove unused namespace

Kjusok replied:
Done

Kjusok committed:
Added the first enemy and his weapon

Added:
Аirst enemy and his weapon
Fixed: enemy taking damage + projectiles hitting player

Kjusok committed:
Fixed: WeaponRoot and MuzzleRoot

Fixed: WeaponRoot and MuzzleRoot

Kjusok committed:
removed unused namespace

removed unused namespace

ega-ega commented on Assets/Scripts/Gameplay.Services/FX/PlayingFX.cs:
```diff
@@ -1,32 +1,36 @@
     {
         private ParticleSystem _particlePrefab;
         private Vector3 _position;
+        private Vector3 _direction;
```
Используй FXContext поле

Kjusok replied:
Done

ega-ega commented on Assets/Scripts/Gameplay/Character/View/CharacterModelRoots.cs:
```diff
@@ -5,7 +5,9 @@ namespace Gameplay
     public class CharacterModelRoots  : MonoBehaviour
     {
         [SerializeField] private Transform _head;
+        [SerializeField] private Transform _rightHand;
```
переименуй в Weapon

Kjusok replied:
Done

ega-ega commented on Assets/Scripts/Gameplay/Projectiles/Projectile.cs:
```diff
@@ -57,8 +58,12 @@ private void OnTriggerEnter(Collider other)
         {
             var target = other.GetComponent<IDamageable>();
```
перенеси и поставь перед IF

Kjusok replied:
Done

ega-ega commented on Assets/Scripts/Gameplay/Weapons/Implementation/Weapon.cs:
```diff
@@ -16,16 +16,16 @@ public class Weapon : MonoBehaviour, IWeapon, ITicker, ICanReload
         private IWeaponSettings _settings;
         private Cooldown _shotCooldown;
         private Cooldown _reloadCooldown;
+        private Transform _muzzleRoot;
```
замени на поле типа WeaponModelRoots

Kjusok replied:
Done

ega-ega commented on Assets/Scripts/Gameplay/Weapons/Implementation/WeaponModelRoots.cs:
```diff
@@ -0,0 +1,11 @@
+{
+    public class WeaponModelRoots : MonoBehaviour
+    {
+        [SerializeField] private Transform _muzzle;
```
[field: SerializeFiled] public Transform Muzzle { get; }

Kjusok replied:
Нужен уточняющий вопрос

Kjusok committed:
Refactoring

Refactoring

Kjusok committed:
Refactoring

Refactoring
